### PR TITLE
Portfolio prod

### PR DESCRIPTION
**Pull request by Obinnabii**
### Style the portfolio page to look like a terminal Console and add contents

There are __three main sections__ namely;
- **_about me_**
- **_languages_**
- **_projects_**

**_about me_** and **_languages_** are rendered in _script.js_. They are typed into the _console_ div in _index.html_ using the _console_ typewriter.
**_projects_** is rendered in _index.html_. the individual projects start off invisible. The **addProjects(**_div_name_**)** function makes them visible. it is integrated into the typewriter calls for callback reasons.

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -4,19 +4,34 @@
...
     <script src="script.js"></script>
   </head>
 
+
   <body>
+
     <div id="content">
+      <header>
+      <!-- Navigation bar at the top -->
+      <nav>
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: Please try to indent these according to the style guide: 2 indents each time we go down a level https://google.github.io/styleguide/htmlcssguide.html#Indentation.

Comment (Obinnabii) [reply portfolio/src/main/webapp/index.html]: Sure thing!

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -1,3 +1,18 @@
+a {
+  color: rgb(0, 180, 0);
+  text-decoration: none;
+}
+
+a:hover{
```
Comment (billyeh) [inline portfolio/src/main/webapp/style.css]: nit: add space before {, here and on line 10.

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (Obinnabii) [reply portfolio/src/main/webapp/index.html]: Is there perhaps an auto-styling add on for cloud shell?

**Comment by Obinnabii** [comment]
Added project blurb skeleton here

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
...
+      <div class='col'><i class='fab fa-python'></i></div>\
+      <div class='col'> Python</div>\
+    </div>\
+  </div>"
+
+  // <div class='col col-2'> \
+  //   C/C++ \
+  // </div>\
+
+const projects_blurb = "<div class = 'row'> <p> Below are some projects that I have worked on.</p></div>\
```
Comment (Obinnabii) [inline portfolio/src/main/webapp/script.js]: added a skeleton for the projects_blurb

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
...
+ <!-- row for project cards -->\
+ <div class='row row-cols-4 justify-content-center text-center emphasis align-middle'>\
+  <!-- card -->\
+  <div class='card' style='width: 18rem;'>\
+    <img src='...' class='card-img-top' alt='...'>\
+    <div class='card-body'>\
+      <p class='card-text'>Some quick example text to build on the card title and make up the bulk of the card's content.</p>\
+    </div>\
+  </div>\
+</div>\
```
Comment (Obinnabii) [inline portfolio/src/main/webapp/script.js]: added a skeleton for the projects_blurb

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -41,6 +41,22 @@
...
+          </div>
+          <div class= "col-7">
+            <p>Built a <span class='emphasis'>retro shooter game</span> in Python under the supervision of Prof. Walker White.
+              The game is designed to vaguely resemble the popular space invaders arcade game from 1978. I used 8-bit images to connote the aliens and the spaceship
+              . The game employed a round system with each round having increased alien speed. The player controls the ship and tries to eliminate all the aliens before his lives run out.
+              I designed extra life and extra laser bolt powerups. This project relied heavily on <span class='emphasis'>the concept of inheritance</span>.
+            </p>
+          </div>
+        </div>
+      </div>
```
Comment (Obinnabii) [inline portfolio/src/main/webapp/index.html]: format changes in later commits, this setup worked at first but needed to be tweaked

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -73,33 +74,38 @@ const language_blurb = "<div class = 'row'> <p> I have experience with the follo
...
-  <div class='card' style='width: 18rem;'>\
-    <img src='...' class='card-img-top' alt='...'>\
-    <div class='card-body'>\
-      <p class='card-text'>Some quick example text to build on the card title and make up the bulk of the card's content.</p>\
-    </div>\
-  </div>\
-</div>\
-"
-
+const projects_blurb = "</div><div class = 'row'> <h2>projects</h2></div><div class = 'row'> <p> Below is a selection of some of my favorite projects.</p>"
```
Comment (Obinnabii) [inline portfolio/src/main/webapp/script.js]: added headings for the language and projects section

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
...
+    </div>\
+    <div class='col col-2 '> \
+      <div class='col'><i class='fab fa-python'></i></div>\
+      <div class='col'> Python</div>\
+    </div>\
+  </div>"
+
+  // <div class='col col-2'> \
+  //   C/C++ \
+  // </div>\
```
Comment (Obinnabii) [inline portfolio/src/main/webapp/script.js]: I couldn't find an icon for C/C++. I tried putting it in without an Icon but the text wouldn't show up properly. I wanted it to be vertically aligned in the center so it wouldn't look weird next to all the other languages. I'm still trying to find a solution to this. I'm open to any suggestions

File: portfolio/src/main/webapp/index.html [hunk 8]
```diff
@@ -3,26 +3,121 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Fira+Code:wght@400;500&family=IBM+Plex+Mono:wght@400;500;700&display=swap" rel="stylesheet">
+    
+    <!--Icons-->
+    <script src="https://kit.fontawesome.com/2e7fd39b04.js" crossorigin="anonymous"></script>
+    <script src="https://code.iconify.design/1/1.0.6/iconify.min.js"></script>
+
+    <!-- Bootstrap CSS only -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk" crossorigin="anonymous">
+    
+    <!-- custom styles -->
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: nit: capitalize Custom

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -3,26 +3,121 @@
...
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+      <!-- Console -->
+      <div id="console"></div>
 
+      <!--projects -->
+      <!--Space invaders-->
+      <div id="project_0" class ="project_wrapper">
+        <div class=" row row-cols-2">
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: nit: remove extra space after "

File: portfolio/src/main/webapp/index.html [hunk 10]
```diff
@@ -3,26 +3,121 @@
...
-       <!-- Greeting generator -->
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+      <!-- Console -->
+      <div id="console"></div>
 
+      <!--projects -->
+      <!--Space invaders-->
+      <div id="project_0" class ="project_wrapper">
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: remove space before =

File: portfolio/src/main/webapp/index.html [hunk 11]
```diff
@@ -3,26 +3,121 @@
...
+            <p>Built a <span class='emphasis'>retro shooter game</span> in <span class='emphasis'>Python</span> under the supervision of Prof. Walker White.
+              The game is designed to vaguely resemble the popular space invaders arcade game from 1978. I used 8-bit images to connote the aliens and the spaceship
+              . The game employed a round system with each round having increased alien speed. The player controls the ship and tries to eliminate all the aliens before his lives run out.
+              I designed extra life and extra laser bolt powerups. This project relied heavily on <span class='emphasis'>the concept of inheritance</span>.
+            </p>
+          </div>
+        </div>
+      </div>
+
+      <!--stock exchange simulator-->
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: nit: capitalize Stock

File: portfolio/src/main/webapp/index.html [hunk 12]
```diff
@@ -3,26 +3,121 @@
...
 
-       <!-- Greeting generator -->
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+      <!-- Console -->
+      <div id="console"></div>
 
+      <!--projects -->
+      <!--Space invaders-->
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: nit: for all comments, please try to follow the style guide style https://google.github.io/styleguide/htmlcssguide.html and original examples in the file i.e.

File: portfolio/src/main/webapp/index.html [hunk 13]
```diff
@@ -3,26 +3,121 @@
...
+      
     </div>
+    
+
+
+
+
+
+
+
```
Comment (billyeh) [inline portfolio/src/main/webapp/index.html]: Let's remove the extra whitespace here and below, lines 118-121.

File: portfolio/src/main/webapp/script.js [hunk 14]
```diff
@@ -12,17 +12,104 @@
...
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+//HELPER_FUNCTION
+/**Create a typewriter from a given div */
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js]: PTAL at JS style guide on comments https://google.github.io/styleguide/htmlcssguide.html.

Similar to HTML comments, want a space after the // and /*

Here and below.

File: portfolio/src/main/webapp/script.js [hunk 15]
```diff
@@ -12,17 +12,104 @@
...
+function typify(div_id){
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js]: nit: add space after ) before {

File: portfolio/src/main/webapp/script.js [hunk 16]
```diff
@@ -12,17 +12,104 @@
...
+  //get the div
+  var div = document.getElementById(div_id);
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js]: let or const https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

File: portfolio/src/main/webapp/script.js [hunk 17]
```diff
@@ -12,17 +12,104 @@
...
 }
+
+// TEXT
+/**Heading text */
+const heading_text = "<div class = 'row'> <h1> Obi Abii's Portfolio </h1> </div>"; 
+
+/**About me blurb */
+const about_me_text = "<div class = 'row'> <p>Hi, My name is Obi and this is my <span class='emphasis'>portfolio! <br></span></p>\
+<p>I'm a CS Major at Cornell, currently interning at Google for the Summer of 2020. My main career focus is <span class='emphasis'>software engineering</span>. \
+I am particularly interested in <span class='emphasis'>Machine learning.</span></p></div>";
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js]: For our HTML in JS, let's try using the template literals in JS? https://google.github.io/styleguide/jsguide.html#features-strings-template-strings. See also https://google.github.io/styleguide/jsguide.html#features-strings-no-line-continuations, no \ continuations.

Below too

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (billyeh) [reply portfolio/src/main/webapp/script.js]: Sorry, seeing this comment now, not sure if the suggestion to try another icon was helpful?

File: portfolio/src/main/webapp/script.js [hunk 18]
```diff
@@ -12,17 +12,104 @@
...
+/**print the initial text blurbs */
+
+console
+  .typeString(heading_text) //print heading
+  .pauseFor(300)
+  .typeString(about_me_text) //print about_me section
+  .pauseFor(300)
+  .typeString(language_blurb) //print language_section
+  .pauseFor(300)
+  .typeString(projects_blurb) //print project header
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js]: I would lean towards removing these "print ... " comments. They don't really say anything the code doesn't already.

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (Obinnabii) [reply portfolio/src/main/webapp/script.js]: it was!

File: portfolio/src/main/webapp/script.js [hunk 19]
```diff
@@ -12,17 +12,149 @@
...
+    <div class="col col-2"> 
+      <div class="col">
+        <span class="iconify" data-icon="cib:coreui-c" data-inline="false"></span>
+      </div>
+      <div class="col"> C/C++</div>
+    </div>
+  </div>`
+
+/**
+* blurb containg information about the languages I know
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js:113]: Blurb containing*

File: portfolio/src/main/webapp/script.js [hunk 20]
```diff
@@ -12,17 +12,149 @@
...
+  <div class="row"> <p> Below is a selection of some of my favorite projects.</p></div>`
+
+
+// DIV_ID
+
+/** 
+* The id of the div where the above text will be placed 
+* @type {String}
+*/
+const console_id = "console";
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js:127]: nit: CONSOLE_ID

File: portfolio/src/main/webapp/script.js [hunk 21]
```diff
@@ -12,17 +12,149 @@
...
+
+/** 
+* List of project_ids 
+* @type {!Array<String>}
+*/
+const projects = ['project_0', 'project_1', 'project_2']
```
Comment (billyeh) [inline portfolio/src/main/webapp/script.js:133]: const PROJECTS

File: portfolio/src/main/webapp/style.css [hunk 22]
```diff
@@ -1,9 +1,70 @@
...
+
+nav {
+  padding-top: 3vh;
+  padding-bottom: 3vh;
+ }
+
+/* PROJECT SECTION */
+
+.project-wrapper {
+  visibility: hidden;
```
Comment (billyeh) [inline portfolio/src/main/webapp/style.css:46]: Let's alphabetize the CSS declarations https://google.github.io/styleguide/htmlcssguide.html#Declaration_Order

**Review by billyeh — APPROVED** [review]
It looks really nice now, thanks for addressing the comments!

**Pull request merged**